app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsModel.kt

```diff
@@ -107,4 +113,58 @@ class AppSettingsModel @Inject constructor(
 
     val pushEndpoint = preference.unifiedPushEndpointFlow()
 
+    private val _pushDistributor = MutableStateFlow<String?>(null)
+    val pushDistributor get() = _pushDistributor.asStateFlow()
+
+    private val _pushDistributors = MutableStateFlow<List<PushDistributorInfo>?>(null)
+    val pushDistributors get() = _pushDistributors.asStateFlow()
+
+    fun updatePushDistributor(pushDistributor: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            UnifiedPush.saveDistributor(context, pushDistributor)
+            UnifiedPush.registerApp(context)
+            _pushDistributor.emit(pushDistributor)
+        }
+    }
+
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
I'd put the lines below into a `private [suspend] fun loadPushDistributors`, which is then integrated into the `// push distributor` section.

(And for instance if we would have tests, we could test loading the push distributors.)